src/consent.ts

```diff
@@ -203,30 +328,45 @@ export default function Consent(mpInstance) {
         },
     };
 
-    this.createConsentState = function(consentState) {
-        var gdpr = {};
-        var ccpa = {};
+    // TODO: Refactor this method into a constructor
+    this.createConsentState = function (
+        this: ConsentState,
+        consentState?: ConsentState
+    ): IConsentState {
+        let gdpr = {};
```

Comment:
Perhaps we should have a note of what the difference between `IConsentState` and `ConsentState` to help out future Alex and Rob.  I see that one has the deprecated method but can't quite figure out when you decide to use `ConsentState` vs `IConsentState`

Reply:
The difference is that `ConsentState` is defined in DefTyped and does not have `removeCCPAState`, but `IConsentState` is a local override to keep `removeCCPAState` since we have not properly deprecated the method.

Reply:
See line 99 of src/consent.ts in this PR.

https://github.com/mParticle/mparticle-web-sdk/pull/342/files#diff-2fe6c689e234643987ca8ef0b2c744e50bc2ab8ab16fd454b31a155fc204abaeR99

Reply:
makes sense